lib/improver/spotdata/common_functions.py

```diff
@@ -0,0 +1,408 @@
+            node_list[k, max_list] = node_list[k, max_list] - coord_max
+        else:
+            node_list = np.delete(node_list,
+                                  np.hstack((min_list, max_list)), 1)
+
+    return node_list.tolist()
+
+
+def get_nearest_coords(cube, latitude, longitude, iname, jname):
+    """
+    Uses the iris cube method nearest_neighbour_index to find the nearest grid
+    points to a given latitude-longitude position.
+
+    Args:
+    -----
+    cube : iris.cube.Cube
+        Cube containing a representative grid.
+
+    latitude/longitude : floats
+        Latitude/longitude coordinates of spot data site of interest.
+
+    iname/jname : strings
+        Strings giving the names of the y/x coordinates to be searched.
+
+    Returns:
+    -------
+    i_latitude/j_latitude : ints
+        Grid coordinates of the nearest grid point to the spot data site.
+
+    """
+    i_latitude = cube.coord(iname).nearest_neighbour_index(latitude)
+    j_longitude = cube.coord(jname).nearest_neighbour_index(longitude)
+    return i_latitude, j_longitude
+
+
+def index_of_minimum_difference(whole_list, subset_list=None):
+    """
+    Returns the index of the minimum value in a list.
+
+    Args:
+    -----
+    whole_list : numpy.array
+        Array to be searched for a minimum value.
+
+    subset_list : numpy.array/None
+        Array of indices to include in the search. If None the entirity of
+        whole_list is searched.
+
+    Returns:
+    --------
+    Index of the minimum value in whole_list.
+
+    """
+    if subset_list is None:
+        subset_list = np.arange(len(whole_list))
+    return subset_list[np.argmin(abs(whole_list[subset_list]))]
+
+
+def list_entry_from_index(list_in, index_in):
+    """
+    Extracts index_in element from each list in a list of lists, and returns
+    as a list.
+    e.g.
+         list_in = [[0,1,2],[5,6,7],[8,9,10]]
+         index_in = 1
+         Returns [1,6,9]
+
+    """
+    return list(zip(*list_in)[index_in])
+
+
+def datetime_constraint(time_in):
+    """
+    Constructs an iris equivalence constraint from a python datetime object.
+
+    Args:
+    -----
+    time_in : datetime.datetime object
+        The time to be used to build an iris constraint.
+
+    Returns: iris.Constraint
+        An iris constraint to be used in extracting data at the given time from
+        a cube.
+
+    """
+    return Constraint(
+        time=PartialDateTime(time_in.year, time_in.month,
+                             time_in.day, time_in.hour))
+
+
+def construct_neighbour_hash(neighbour_finding):
+    """
+    Constructs a hash from the various neighbour finding options. This is used
+    to avoid repeating the same neighbour search more than once.
+
+    Args:
+    -----
+    neighbour_finding : dict
+        A dictionary containing the method, vertical_bias, and land_constraint
+        options for neighbour finding.
+
+    Returns:
+    --------
+    neighbour_hash : string
```

Comment:
Sounds like a slightly cannibalistic breakfast food, but I agree, it's not being returned.